tests/test_cli.py

```diff
@@ -5,20 +5,55 @@
 from .common import get_test_data
 
 
-def test_generate_grid_weights():
+def test_generate_grid_weights_with_nc_input_and_2d_coords():
     runner = CliRunner()
-    ds = map(
-        str,
-        [
-            get_test_data("raven-routing-sample", "finalcat_hru_info.zip")[0],
-            get_test_data("raven-routing-sample", "VIC_streaminputs.nc")[0],
-        ],
-    )
+    params = [
+        get_test_data("raven-routing-sample", "VIC_streaminputs.nc")[0],
+        get_test_data("raven-routing-sample", "finalcat_hru_info.zip")[0],
+    ]
+    params = map(str, params)
 
-    result = runner.invoke(generate_grid_weights, ds)
+    result = runner.invoke(generate_grid_weights, params)
 
     assert result.exit_code == 0
     assert not result.exception
     assert ":NumberHRUs 51" in result.output
     assert ":NumberGridCells 100" in result.output
     assert len(result.output.split("\n")) == 216
+
+
+def test_generate_grid_weights_with_nc_input_and_1d_coords():
+    runner = CliRunner()
+    params = [
+        get_test_data("raven-routing-sample", "era5-test-dataset-crop.nc")[0],
+        get_test_data("raven-routing-sample", "finalcat_hru_info.zip")[0],
+        "--var-names",
+        "longitude",
+        "latitude",
+    ]
+    params = map(str, params)
+
+    result = runner.invoke(generate_grid_weights, params)
+
+    assert result.exit_code == 0
+    assert not result.exception
+    assert ":NumberHRUs 51" in result.output
+    assert ":NumberGridCells 9801" in result.output
+    assert len(result.output.split("\n")) == 128
+
+
+def test_generate_grid_weights_with_shp_input():
```

Comment:
This test runs with the default `AREA_ERROR_THRESHOLD=0.05` which will result in several subbasins where weights do not sum up to 1.

I think it would be good to have a test with having `AREA_ERROR_THRESHOLD=0.5` and checking that one of those basins is adjusted correctly.

Reply:
See: "test_generate_grid_weights_with_weight_rescaling()"